src/VisualStudio/Core/Def/Implementation/Classification/ViewSupportsClassificationService.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using Microsoft.CodeAnalysis.Editor;
+using Microsoft.CodeAnalysis.Editor.Implementation.Classification;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+using Microsoft.VisualStudio.LanguageServices.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.Classification
+{
+    [Export(typeof(IViewSupportsClassificationService))]
+    internal class ViewSupportsClassificationService : IViewSupportsClassificationService
+    {
+        private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
+        private readonly SVsServiceProvider _serviceProvider;
+
+        [ImportingConstructor]
+        public ViewSupportsClassificationService(
+            ITextBufferAssociatedViewService viewService,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService, 
+            SVsServiceProvider serviceProvider)
+        {
+            _editorAdaptersFactoryService = editorAdaptersFactoryService;
+            _serviceProvider = serviceProvider;
+        }
+
+        public bool CanClassifyView(ITextView view)
+        {
+            var vsTextView = _editorAdaptersFactoryService.GetViewAdapter(view);
+            if (vsTextView == null)
+            {
+                return true;
+            }
+
+            return !vsTextView.IsImmediateWindow(
+                (IVsUIShell)_serviceProvider.GetService(typeof(SVsUIShell)),
+                immediateWindowTextView => view == _editorAdaptersFactoryService.GetWpfTextView(immediateWindowTextView));
```

Comment:
Why the crazy lambda here? You've already passed in the IVsTextView, which has a nice 1:1 mapping to IWpfTextView.